python/tank/util/metrics.py

```diff
@@ -241,6 +223,18 @@ def __init__(self, engine):
     def run(self):
         """Runs a loop to dispatch metrics that have been logged."""
 
+        # first of all, check if metrics dispatch is supported
+        # connect to shotgun and probe for server version
+        sg_connection = self._engine.shotgun
+        metrics_ok = (
```

Comment:
can you elaborate a bit on why this needs to be here rather than short circuiting at the dispatcher level? just want to make sure i understand fully. :)